pyproject.toml

```diff
@@ -1,24 +1,105 @@
-# Example configuration for Black.
-
-# see https://raw.githubusercontent.com/ambv/black/master/pyproject.toml
-
-[tool.black]
-line-length = 88
-include = '\.pyi?$'
-exclude = '''
-/(
-    \.git
-  | \.hg
-  | \.mypy_cache
-  | \.tox
-  | \.venv
-  | \.ve
-  | _build
-  | buck-out
-  | build
-  | dist
-  | migrations
-  | env
-  | ve
-)/
-'''
+[tool.poetry]
+name = "rp-sidekick"
+version = "1.11.7"
+description = ""
+authors = ["Praekelt Foundation <dev@praekeltfoundation.org>"]
+repository = "http://github.com/praekeltfoundation/rp-sidekick"
+license = "BSD"
+readme = "README.rst"
+
+packages = [
+  { include = "config" },
+  { include = "msisdn_utils" },
+  { include = "randomisation" },
+  { include = "rp_dtone"},
+  { include = "rp_interceptors"},
+  { include = "rp_recruit"},
+  { include = "rp_transferto"},
+  { include = "rp_yal"},
+  { include = "sidekick" },
+]
+
+[tool.poetry.dependencies]
+python = "^3.9"
+celery = "^5.2.3"
+django = "^4.2.15"
```

Comment:
In `setup.py` we had all these requirements pinned to the exact version where the `^` here might install a later minor or patch version. Was this intended?

Reply:
Yes. Pinning to exact versions makes it hard to tell which things are pinned because they need to be and which are pinned because it's easier to pin everything. Poetry's lockfile specifies the exact versions of all the dependencies (including transitive dependencies, which have caused us problems in the past) in a much cleaner way.

Reply:
Oh ok, so `poetry install` will always install what we have in the lock file, until we run `poetry update` which will update the lock file within the constraints we defined here.